### PR TITLE
Fixes #24111: Memory leak in crypto/conf/conf_mod.c

### LINKED ISSUE #24111: There is a memory leak defect on line 523 of the file /openssl/crypto/conf/conf_mod.c.
On line 510 of the file /openssl/crypto/conf/conf_mod.c, a pointer variable named 'to_delete' is defined. This pointer variable is allocated a dynamic memory area through the function sk_CONF_MODULE_new_null on line 519. When the if statement on line 521 returns true, the program will return on line 523, thereby bypassing the release operation on the dynamic memory area pointed to by the 'to_delete' pointer on line 546. This constitutes a memory leak defect, as shown in the diagram below:
https://github.com/LuMingYinDetect/openssl_defects/blob/main/openssl_13.png

### PR DESCRIPTION
**Pull request by coolshrid**
Moved the statement from line 520 to line 525, so we only allocate after the if loop that possibly can return

**Before**
```c
    to_delete = sk_CONF_MODULE_new_null(); 

    if (new_modules == NULL) {
        ossl_rcu_write_unlock(module_list_lock);
        return;
    }

```

**After**
```c
    // to_delete = sk_CONF_MODULE_new_null(); 

    if (new_modules == NULL) {
        ossl_rcu_write_unlock(module_list_lock);
        return;
    }

    to_delete = sk_CONF_MODULE_new_null();
```

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -256,3 +256,6 @@ doc-nits
 # LSP (Language Server Protocol) support
 .cache/
 compile_commands.json
+**/.DS_Store
+.vscode/settings.json
+.vscode/tasks.json
```
Comment (t8m) [inline .gitignore:261]: Please do not add unrelated changes to a commit. If you want them here in the PR (although it would better be a separate PR), please at least separate them to a different commit.

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by nhorman** [comment]
@coolshrid this pr is waiting on you to make some requested changes.  If you do not respond we will have to close this PR

**Comment by coolshrid** [comment]
Fixed

**Comment by t8m** [comment]
Please drop the merge commit. `git reset --hard HEAD~ ; git push --force` should help.

**Comment by coolshrid** [comment]
Done

File: .gitignore [hunk 2]
```diff
@@ -255,4 +255,4 @@ doc-nits
 
 # LSP (Language Server Protocol) support
 .cache/
-compile_commands.json
+compile_commands.json
```
Comment (t8m) [inline .gitignore:258]: Please add back the end of line character.

Comment (coolshrid) [reply .gitignore]: Done

**Review by tom-cosgrove-arm — APPROVED** [review]

**Comment by t8m** [comment]
Squashed and merged to the master and 3.3 branches. Thank you for your contribution.
